hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -179,9 +183,13 @@ bool relocateSmashable(Env& env, Instruction* src, Instruction* dest,
    * jmp that can be optimized. It's important to not simply analyze this
    * smashable jmp without first considering if it's part of a jcc.
    */
-  auto target = smashableJmpTarget(srcAddr);
+  auto target = smashableJmpTarget(srcAddrActual);
   if (target) {
-    auto sl = getSmashableFromTargetAddr(srcAddr + kSmashJmpTargetOff);
+    // Look up the smashable (jmp, jcc, etc) from the target. Note this is
+    // an actual address, and so we'll use the offset with the (potentially)
+    // virtual srcAddr when we search through smashableLocation.
+    auto slActual = getSmashableFromTargetAddr(srcAddrActual + kSmashJmpTargetOff);
+    auto sl = srcAddr + (slActual - srcAddrActual);
```

Comment:
I'd prefer either having getSmashableFromTargetAddr take an actual and a virtual address or adding a function to CodeBlock to do the reverse mapping, this is a little confusing.